ray-operator/controllers/ray/rayjob_controller_scheduled_test.go

```diff
@@ -0,0 +1,175 @@
+			underlyingK8sJob := &batchv1.Job{}
+			Consistently(
+				// k8sClient client throws error if resource not found
+				func() bool {
+					err := getResourceFunc(ctx, common.RayJobK8sJobNamespacedName(rayJob), underlyingK8sJob)()
+					return errors.IsNotFound(err)
+				},
+				time.Second*3, time.Millisecond*500).Should(BeTrue())
+		})
+
+		It("should be able to update all Pods to Running", func() {
+			updateHeadPodToRunningAndReady(ctx, rayJob.Status.RayClusterName, namespace)
+			updateWorkerPodsToRunningAndReady(ctx, rayJob.Status.RayClusterName, namespace)
+		})
+
+		It("Dashboard URL should be set", func() {
+			Eventually(
+				getDashboardURLForRayJob(ctx, rayJob),
+				time.Second*3, time.Millisecond*500).Should(HavePrefix(rayJob.Name), "Dashboard URL = %v", rayJob.Status.DashboardURL)
+		})
+
+		It("should create the underlying Kubernetes Job object", func() {
+			underlyingK8sJob := &batchv1.Job{}
+			// The underlying Kubernetes Job should be created when the RayJob is created
+			Eventually(
+				getResourceFunc(ctx, common.RayJobK8sJobNamespacedName(rayJob), underlyingK8sJob),
+				time.Second*3, time.Millisecond*500).Should(Succeed(), "Expected Kubernetes job to be present")
+		})
+
+		It("should transition to the Running", func() {
+			Eventually(
+				getRayJobDeploymentStatus(ctx, rayJob),
+				time.Second*5, time.Millisecond*500).Should(Equal(rayv1.JobDeploymentStatusRunning),
+				"JobDeploymentStatus should be Running")
+		})
+
+		It("RayJobs's JobDeploymentStatus transitions to Scheduled after Complete.", func() {
+			// Update fake dashboard client to return job info with "Succeeded" status.
+			getJobInfo := func(context.Context, string) (*utils.RayJobInfo, error) { //nolint:unparam // This is a mock function so parameters are required
+				return &utils.RayJobInfo{JobStatus: rayv1.JobStatusSucceeded, EndTime: uint64(time.Now().UnixMilli())}, nil
+			}
+			fakeRayDashboardClient.GetJobInfoMock.Store(&getJobInfo)
+			defer fakeRayDashboardClient.GetJobInfoMock.Store(nil)
+
+			// RayJob transitions to Complete if and only if the corresponding submitter Kubernetes Job is Complete or Failed.
+			Consistently(
+				getRayJobDeploymentStatus(ctx, rayJob),
+				time.Second*3, time.Millisecond*500).Should(Equal(rayv1.JobDeploymentStatusRunning), "JobDeploymentStatus = %v", rayJob.Status.JobDeploymentStatus)
+
+			// Update the submitter Kubernetes Job to Complete.
+			namespacedName := common.RayJobK8sJobNamespacedName(rayJob)
+			job := &batchv1.Job{}
+			err := k8sClient.Get(ctx, namespacedName, job)
+			Expect(err).NotTo(HaveOccurred(), "failed to get Kubernetes Job")
+
+			// Update the submitter Kubernetes Job to Complete.
+			conditions := []batchv1.JobCondition{
+				{Type: batchv1.JobComplete, Status: corev1.ConditionTrue},
+			}
+			job.Status.Conditions = conditions
+			Expect(k8sClient.Status().Update(ctx, job)).Should(Succeed())
+
+			// RayJob transitions to Scheduled.
```

Comment:
Per my previous comment https://github.com/ray-project/kuberay/pull/3836#discussion_r2229707551

The validation for transitioning to Scheduled state should be done in a separate IT block